frontend/app/player/MessageDistributor/managers/AssistManager.ts

```diff
@@ -72,6 +72,41 @@ export const INITIAL_STATE: State = {
 
 const MAX_RECONNECTION_COUNT = 4;
 
+let handlers = Symbol('handlers');
+
+interface VideoFeeds {
+  [key: string]: boolean
+}
+
+type ProxyResult<T extends {}> = T & {
+  observe: (handler: (args: any) => any) => void
+  [key: typeof handlers]: Function[]
+}
+
+function makeObservable<T extends {}>(target: T): ProxyResult<T> {
+  // @ts-ignore
+  target[handlers] = [];
+
+  // @ts-ignore
+  target.observe = function(handler) {
+    this[handlers].push(handler);
+  };
+
+  // @ts-ignore
+  return new Proxy(target, {
+    set(target, property, value) {
+      // @ts-ignore
+      let success = Reflect.set(...arguments);
+      if (success) {
+        // @ts-ignore
+        target[handlers].forEach(handler => handler(property, value));
+      }
+      return success;
+    }
+  });
+}
+
+export const videoFeeds = makeObservable<VideoFeeds>({})
```

Comment:
Why?

Reply:
let's better make a special SaaS service for this! it will store the state of each track in a Postgres database and send it back through a socket if the state has changed (triggered with Notifications)!

Reply:
> why

Because I don't want another redux action with state selector when we're trying to go away from redux

> let's better make a special SaaS service for this

Time to start